### PR TITLE
SBS-5698: add support for geospatial feature type partitions

### PR DESCRIPTION
**Pull request by vincent-cognite**

### CODE REVIEW

File: cognite/experimental/_api/geospatial.py [hunk 1]
```diff
@@ -52,6 +59,101 @@ def __init__(self, *args, **kwargs):
                 wrapped = _with_cognite_domain(getattr(GeospatialAPI, attr_name))
                 setattr(ExperimentalGeospatialAPI, attr_name, wrapped)
 
+    @_with_cognite_domain
+    def create_feature_types(
```
Comment (vincent-cognite) [inline cognite/experimental/_api/geospatial.py:63]: in order to use the new definition with partitions for FeatureType, I don't see another way to repeat the implementation for create_feature_types(), list_feature_types() and retrieve_feature_types()
Do you know a better approach @tuanng-cognite ?

Comment (tuanng-cognite) [reply cognite/experimental/_api/geospatial.py:63]: I don't see any other ways either. The point is that the create_feature_types changes, receives another input object.

File: cognite/experimental/_api/geospatial.py [hunk 2]
```diff
@@ -52,6 +59,101 @@ def __init__(self, *args, **kwargs):
...
+
+    @overload
+    def retrieve_feature_types(self, external_id: str) -> FeatureType:
+        ...
+
+    @overload
+    def retrieve_feature_types(self, external_id: List[str]) -> FeatureTypeList:
+        ...
+
+    def retrieve_feature_types(self, external_id: Union[str, List[str]]) -> Union[FeatureType, FeatureTypeList]:
```
Comment (tuanng-cognite) [inline cognite/experimental/_api/geospatial.py:131]: with_domain annotation here?

File: cognite/experimental/_api/geospatial.py (hunk 1 above)
Comment (vincent-cognite) [reply cognite/experimental/_api/geospatial.py:62]: if I keep using the "main" implementations for list_feature_types() and retrieve_feature_types(), I still get the `partitions` field, so I don't need to reimplement them.
Only necessary for create_feature_types() afais

**Review by tuanng-cognite — APPROVED** [review]

**Pull request merged**